### PR TITLE
sidecar for envoy pod to keep IAP up to date

### PR DESCRIPTION
**Pull request by danisla**
Converts the IAP initContainer to a sidecar that configures IAP and monitors for changes to the backend service.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/552)

### CODE REVIEW

File: kubeflow/core/iap.libsonnet [hunk 1]
```diff
@@ -156,14 +156,25 @@
...
+              }),
               {
                 name: "iap",
                 image: "google/cloud-sdk:alpine",
                 command: [
                   "sh",
-                  "/var/envoy-config/iap-init.sh",
-                ],
+                  "/var/envoy-config/iap-init.sh"
+            ],
```
Comment (ankushagarwal) [inline kubeflow/core/iap.libsonnet]: nit: can you format the jsonnet files `./scripts/autoformat_jsonnet.sh`

**Comment by danisla** [comment]
I ran the autoformat script. It cleaned up a bunch of other files, you can see that in my second commit.

**Pull request merged**